pl_examples/domain_templates/computer_vision_fine_tuning.py

```diff
@@ -0,0 +1,440 @@
+
+        epoch = self.current_epoch
+        if epoch < self.hparams.milestones[0] and mode:
+            # feature extractor is frozen (except for BatchNorm layers)
+            freeze(module=self.feature_extractor,
+                   train_bn=self.hparams.train_bn)
+
+        elif self.hparams.milestones[0] <= epoch < self.hparams.milestones[1] and mode:
+            # Unfreeze last two layers of the feature extractor
+            freeze(module=self.feature_extractor,
+                   n=-2,
+                   train_bn=self.hparams.train_bn)
+
+    def on_epoch_start(self):
+        """Use `on_epoch_start` to unfreeze layers progressively."""
+        optimizer = self.trainer.optimizers[0]
+        if self.current_epoch == self.hparams.milestones[0]:
+            _unfreeze_and_add_param_group(module=self.feature_extractor[-2:],
+                                          optimizer=optimizer,
+                                          train_bn=self.hparams.train_bn)
+
+        elif self.current_epoch == self.hparams.milestones[1]:
+            _unfreeze_and_add_param_group(module=self.feature_extractor[:-2],
+                                          optimizer=optimizer,
+                                          train_bn=self.hparams.train_bn)
+
+    def training_step(self, batch, batch_idx):
+
+        # 1. Forward pass:
+        x, y = batch
+        y_logits = self.forward(x)
+        y_true = y.view((-1, 1)).type_as(x)
+        y_bin = torch.ge(y_logits, 0)
+
+        # 2. Compute loss & accuracy:
+        train_loss = self.loss(y_true, y_logits)
+        num_correct = torch.eq(y_bin.view(-1), y_true.view(-1)).sum().item()
+
+        # 3. Outputs:
+        tqdm_dict = {'train_loss': train_loss}
+        output = OrderedDict({'loss': train_loss,
+                              'num_correct': num_correct,
+                              'log': tqdm_dict,
+                              'progress_bar': tqdm_dict})
+
+        return output
+
+    def training_epoch_end(self, outputs):
+        """Compute and log training loss and accuracy at the epoch level."""
+
+        train_loss_mean = torch.stack([output['loss']
+                                       for output in outputs]).mean()
+        train_acc_mean = torch.stack([output['num_correct']
+                                      for output in outputs]).sum().float()
+        train_acc_mean /= (len(outputs) * self.hparams.batch_size)
+        return {'log': {'train_loss': train_loss_mean,
+                        'train_acc': train_acc_mean,
+                        'step': self.current_epoch}}
+
+    def validation_step(self, batch, batch_idx):
+
+        # 1. Forward pass:
+        x, y = batch
+        y_logits = self.forward(x)
+        y_true = y.view((-1, 1)).type_as(x)
+        y_bin = torch.ge(y_logits, 0)
+
+        # 2. Compute loss & accuracy:
+        val_loss = self.loss(y_true, y_logits)
+        num_correct = torch.eq(y_bin.view(-1), y_true.view(-1)).sum().item()
```

Comment:
```suggestion
        num_correct = torch.eq(y_bin.view(-1), y_true.view(-1)).sum()
```
Sorry to be pedantic, but it gives an error now that we are using torch.stack in epoch end, because stacking a list of ints doesn't work. Let's keep it a tensor here? same applies in training.